### DISCUSSION TITLE
Handle libraries that use $ in function names

### DISCUSSION THREAD

**Discussion by Talinx** [QwikDev/qwik-evolution#194] [Proposals For Qwik]
### What is it about?

Add options to the optimizer to handle libraries that use $ in function names without intending this as an instruction for the Qwik optimizer.

#### Problems you are trying to solve:
* Use a library that uses `$` in function names

#### Goals you are trying to achieve:
* Allow the usage of libraries that use `$` in function names

#### Any other context or information you want to share:
I want to use the peggy library (https://peggyjs.org/). This library uses $ in function names, e. g. `peg$SyntaxError`. This causes errors in Qwik, e. g. `Error: Value cannot be serialized in _.SyntaxError, because it's a function named "peg$SyntaxError". You might need to convert it to a QRL using $(fn): const peg$SyntaxError =...`

The Qwik documentation states that it converts `$` suffixes. Here it is not a suffix, yet it causes this error. (I guess `$` as suffix is only the suggested style.) Maybe this is a bug in the optimizer and it should only treat `$` as a special symbol when it is a suffix of the function name?

---

#### What do you propose?
The Qwik optimizer should be able to handle cases where `$` is used in libraries that do not intend this as a special symbol for the optimizer.  A solution could be one of these options:

* Do not treat `$` as a special symbol in library code
* Have an option to add exceptions, e. g. that the `$` symbol should not be treated as a special symbol for certain libraries or functions, e. g. specified by a Regex.

---

### Links / References

A library that uses `$` in function names: https://peggyjs.org/

**Comment by shairez** [comment]
thanks @Talinx !
this sounds more like a bug than a feature

Can you please copy this content into a newly created "bug" issue on the main repo?

Please ping me on discord afterwards, I'll make sure we give it proper attention
thanks!

**Comment by Talinx** [comment]
Thanks, added a corresponding bug report issue: https://github.com/QwikDev/qwik/issues/7091

**Reply by shairez** [reply]
Thank you @Talinx !
Added it to our kanban with a priority
